pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -184,17 +227,79 @@ nodesLoop:
 	return res, nil
 }
 
+func (r *Runtime) getTargets(ctx context.Context, params *params.Params) ([]target, error) {
+	switch r.connectionMode {
+	case ConnectionModeKubernetesProxy:
+		// Get nodes to run on
+		nodes := params.Get(ParamNode).AsStringSlice()
+		pods, err := getGadgetPods(ctx, nodes)
+		if err != nil {
+			return nil, fmt.Errorf("get gadget pods: %w", err)
+		}
+		if len(pods) == 0 {
+			return nil, fmt.Errorf("get gadget pods: Inspektor Gadget is not running on the requested node(s): %v", nodes) //nolint:all
+		}
+		return pods, nil
+	case ConnectionModeDirect:
+		inTargets := params.Get(ParamRemoteAddress).AsStringSlice()
+		targets := make([]target, 0)
+		for _, t := range inTargets {
+			purl, err := url.Parse(t)
+			if err != nil {
+				return nil, fmt.Errorf("invalid remote address %q: %w", t, err)
+			}
+			targets = append(targets, target{
+				name: t,
+				node: purl.Hostname(),
```

Comment:
Yeah, that's true for local unix socket connections (`unix:///path/to/socket` is missing a hostname); if you'd use tcp, this would be the IP or hostname. Should we use a hardcoded "localhost" when hostname is empty?

Reply:
This target works very well for kubernetes, there you have a pod and node, but for connection mode direct it's confusing. I'll say to leave the node empty and to udpate all debug / error messages to only include it if it's not empty.

Reply:
I've changed it so that for unix sockets it'll now say "local" and for tcp targets it will be the target hostname.